core/src/main/scala/org/zalando/grafter/Rewriter.scala

```diff
@@ -90,7 +96,7 @@ trait Rewriter {
       new scala.collection.mutable.HashMap[String, Any]
 
     strategy[Any] {
-      case v if predicate(v) =>
+      case v if predicate(v) && canBeRewritten(v) =>
```

Comment:
We don't even try to make singletons for things like `java.lang.String` because it will return crazy results (the same string in all configurations needing a string for example).